File: frontend/src/types/property.ts
export interface Property {
  id: number;
  title: string;
  description: string;
  price: number;
  address: string;
  city: string;
  state: string;
  zip_code: string;
  bedrooms: number;
  bathrooms: number;
  square_footage: number;
  property_type: string;
  listing_status: string;
  photos: string[];
  latitude: number | null;
  longitude: number | null;
}

export interface PropertyFilters {
  min_price?: number;
  max_price?: number;
  city?: string;
  bedrooms?: number;
  property_type?: string;
  search?: string;
}

export interface ContactFormData {
  property_id: number;
  name: string;
  email: string;
  phone?: string;
  message: string;
}

File: backend/app/schemas/property.py
from typing import Optional
from pydantic import BaseModel


class PropertyOut(BaseModel):
    id: int
    title: str
    description: str
    price: int
    address: str
    city: str
    state: str
    zip_code: str
    bedrooms: int
    bathrooms: float
    square_footage: int
    property_type: str
    listing_status: str
    photos: list[str]
    latitude: Optional[float] = None
    longitude: Optional[float] = None

    class Config:
        from_attributes = True


class PropertyFilter(BaseModel):
    min_price: Optional[int] = None
    max_price: Optional[int] = None
    city: Optional[str] = None
    state: Optional[str] = None
    bedrooms: Optional[int] = None
    property_type: Optional[str] = None
    search: Optional[str] = None
